framework/src/parser/Parser.C

```diff
@@ -139,425 +238,320 @@ Parser::getFileName(bool stripLeadingPath) const
+      {
+        mooseError("invalid hit in arg '", arg, "': ", err.what());
+      }
+    }
+  }
+  return hit_text;
 }
 
 void
-Parser::checkUnidentifiedParams(std::vector<std::string> & all_vars,
-                                bool error_on_warn,
-                                bool in_input_file,
-                                std::shared_ptr<FEProblemBase> fe_problem) const
+Parser::parse(const std::string & input_filename)
 {
-  // Make sure that multiapp overrides were processed properly
-  int last = all_vars.size() - 1;                // last is allowed to go negative
-  for (int i = 0; i <= last; /* no increment */) // i is an int because last is an int
-  {
-    std::string multi_app, variable;
-    int app_num;
-
-    /**
-     * Command line parameters that contain a colon are assumed to apply to MultiApps
-     * (e.g.  MultiApp_name[num]:fully_qualified_parameter)
-     *
-     * Note: Two separate regexs are used since the digit part is optional. Attempting
-     * to have an optional capture into a non-string type will cause pcrecpp to report
-     * false. Capturing into a string an converting is more work than just using two
-     * regexs to begin with.
-     */
-    if (pcrecpp::RE("(.*?)"  // Match the MultiApp name
-                    "(\\d+)" // MultiApp number (leave off to apply to all MultiApps with this name)
-                    ":"      // the colon delimiter
-                    "(.*)"   // the variable override that applies to the MultiApp
-                    )
-            .FullMatch(all_vars[i], &multi_app, &app_num, &variable) &&
-        fe_problem->hasMultiApp(multi_app) && // Make sure the MultiApp exists
-        // Finally make sure the number is in range (if provided)
-        static_cast<unsigned int>(app_num) < fe_problem->getMultiApp(multi_app)->numGlobalApps())
-
-      // delete the current item by copying the last item to this position and decrementing the
-      // vector end position
-      all_vars[i] = all_vars[last--];
-    else if (pcrecpp::RE("(.*?)" // Same as above without the MultiApp number
-                         ":"
-                         "(.*)")
-                 .FullMatch(all_vars[i], &multi_app, &variable) &&
-             fe_problem->hasMultiApp(
-                 multi_app)) // Make sure the MultiApp exists but no need to check numbers
-
-      // delete (see comment above)
-      all_vars[i] = all_vars[last--];
-
-    // TODO: check to see if globals are unused
-    else if (all_vars[i].find(":") == 0)
-      all_vars[i] = all_vars[last--];
-
-    else
-      // only increment if we didn't "delete", otherwise we'll need to revisit the current index
-      // due
-      // to copy
-      ++i;
-  }
-
-  mooseAssert(last + 1 >= 0, "index \"last\" is negative");
-
-  // Remove the deleted items
-  all_vars.resize(last + 1);
-
-  std::set<std::string> difference;
-
-  std::sort(all_vars.begin(), all_vars.end());
+  // Save the filename
+  char abspath[PATH_MAX + 1];
```

Comment:
I don't see where PATH_MAX is defined, but it looks like a compile time constant. You could use `std::array` here. We generally avoid raw c-arrays in MOOSE.

Reply:
the variable is just a temporary to pass to a c stdlib API.  So using std::array wouldn't really buy anything except making the code have an extra ``.data()`` call when I pass in the var to ``realpath``.  PATH_MAX is a stdlib constant.

Reply:
Yes, I realize it's a local. I'm just saying that there are new ways to create stack allocated arrays. It's about style, not function. This is ok here, it's very limited scope.